app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,20 @@
+class AuthController < ApplicationController
+  def log_in; end
+
+  def callback
+    decoded_token = AuthService.decode_token(params['firebase_token'])
+    @user_name = decoded_token[0]['name']
+    @user_id = decoded_token[0]['user_id']
+    @user_email = decoded_token[0]['email']
+
+    user = User.where(display_name: @user_name, uid: @user_id, email: @user_email).first_or_create
+    session[:user] = user.id if user
+
+    render json: session[:user]
```

Comment:
Will this be proper JSON? Don't you need to wrap it, e.g. `{ 'user': 1 }`?

Reply:
Fixed the JSON by returning the user model and a forwarding url if one exists